config.mk

```diff
@@ -24,6 +24,18 @@ CPPFLAGS += -I$(INC)
 # -std=c99 may not be supported by some very old compilers...
 CFLAGS ?= -g -Os -Wall
 
+# Grab the system information
+UNAME_S := $(shell uname -s)
+
+# Set the shared library extension
+ifeq ($(UNAME_S),Darwin)
+  SOEXT := dylib
+  $(info INFO: detected MacOS)
+else
+  SOEXT := so
+  $(info INFO: detected Linux)
```

Comment:
I would not say Linux here. Maybe 'standard Unix`? Or just not say anything for the non-macos case? (maybe the latter is better).

Reply:
I'll just remove the prints. I had mostly forgotten I put them in there while I was debugging.